src/services/reportCache.js

```diff
@@ -162,21 +162,28 @@ export const cacheCitations = async (objectiveId, activityReportObjectiveId, cit
     hookMetadata: { objectiveId },
   });
 
+  // Get the goal for this objective.
+  const goal = await Goal.findOne({
+    attributes: ['grantId', 'createdVia'],
+    include: [
+      {
+        model: Objective,
+        as: 'objectives',
+        where: { id: objectiveId },
+        required: true,
+      },
+    ],
+  });
+
+  if (goal && goal.createdVia !== 'monitoring') {
+    // If this is no longer a monitoring goal associated with this objective,
+    // we don't (and shouldn't) save any citations.
+    return [];
+  }
+
```

Comment:
What happens if the `goal `is `null `here?

We might want to add something like this:

```
if (!goal) {
  auditLogger.info(`No goal found for objective ${objectiveId}. Skipping citation caching.`);
  return [];
}
```

Reply:
I added to the existing if statement.